src/scores/spatial/fss_impl.py

```diff
@@ -0,0 +1,371 @@
+"""
+This module contains methods related to the FSS score.
+
+For an explanation of the FSS, and implementation considerations, see references below.
+
+Currently uses the default score defined Robert and Leans (2008) :sup:`[1,2]`. Fine grained controls are considered in `#353 <GITHUB353_>`_.
+
+The default computation is performed using`numpy` :sup:`[3]` and summed-area tables, with future optimization options considered in:
+- `#269 <GITHUB269_>`_.
+- `#270 <GITHUB270_>`_.
+
+References:
+1. https://journals.ametsoc.org/view/journals/mwre/136/1/2007mwr2123.1.xml
+2. https://journals.ametsoc.org/view/journals/mwre/149/10/MWR-D-18-0106.1.xml
+3. https://www.researchgate.net/publication/269222763_Fast_calculation_of_the_Fractions_Skill_Score
+
+.. _GITHUB269: https://github.com/nci/scores/issues/269
+.. _GITHUB270: https://github.com/nci/scores/issues/270
+.. _GITHUB353: https://github.com/nci/scores/issues/353
+"""
+import warnings
+from typing import Callable, Optional, Tuple
+
+import numpy as np
+import numpy.typing as npt
+import xarray as xr
+
+from scores.fast.fss.fss_backends import get_compute_backend
+from scores.fast.fss.typing import FssComputeMethod, FssDecomposed
+from scores.typing import FlexibleDimensionTypes, XarrayLike
+from scores.utils import (
+    DimensionError,
+    DimensionWarning,
+    FieldTypeError,
+    NumpyThresholdOperator,
+    gather_dimensions,
+    left_identity_operator,
+)
+
+
+def fss_2d(  # pylint: disable=too-many-locals,too-many-arguments
+    fcst: xr.DataArray,
+    obs: xr.DataArray,
+    *,  # Force keywords arguments to be keyword-only
+    event_threshold: np.float64,
+    window_size: Tuple[int, int],
+    spatial_dims: Tuple[str, str],
+    zero_padding: bool = False,
+    reduce_dims: Optional[FlexibleDimensionTypes] = None,
+    preserve_dims: Optional[FlexibleDimensionTypes] = None,
+    threshold_operator: Callable = np.greater,
+    compute_method: FssComputeMethod = FssComputeMethod.NUMPY,
+    dask: str = "forbidden",  # see: `xarray.apply_ufunc` for options
+) -> xr.DataArray:
+    """
+    Uses :py:func:`fss_2d_single_field` to compute the fraction skills score for each 2D spatial
```

Comment:
Is this meant to be either "Fractions Skill Score" or "fractions skill score"? If this instance is meant to be "fraction skills score", then please ignore this feedback.

Reply:
Thanks, it should be Fraction**s** Skill Score